### DISCUSSION TITLE
Getting realtime errors after few minutes of idle on web

### DISCUSSION THREAD

**Discussion by cRaZy92** [pocketbase/pocketbase#642] [Q&A] [answered]
Hello,
I am getting request errors while I am not doing anything on my web that is using Pocketbase. 
There are 2 real-time connections active and after a while, I get errors like this: 
![image](https://user-images.githubusercontent.com/33964772/192170591-99f7b5fa-9b6d-4aad-98fa-434b0ee09d1a.png)
There is no further information, when I click on the error, it redirects me to a specific request:
![image](https://user-images.githubusercontent.com/33964772/192170677-954dcc7f-ce86-46d1-af49-919227034903.png)
![image](https://user-images.githubusercontent.com/33964772/192170742-2523261b-655b-4d98-8c27-3ff5672c9d58.png)
![image](https://user-images.githubusercontent.com/33964772/192170790-395a0016-474a-49d5-86dd-9c6b07b3bf99.png)

It's not really broken but it gets old with that error in the console like 100s times one after another.
I am not sure what else I can provide, please let me know.

Thanks for your help!

**Comment by ganigeorgiev** [comment, marked as answer]
Hm, maybe there is a fly.io limit of how long a http connection could remain open. I haven't really tested the realtime api with Fly.io. Will try to find some time later today or tomorrow and will see if I can reproduce it.

In any case, there should be an auto reconnect in place, so even if there is a limit, you can ignore it and the SDK will try to keep the realtime connection alive.

**Reply by ganigeorgiev** [reply]
There is a 60sec limit for long-lived connections on fly.io - https://community.fly.io/t/long-lived-tcp-connections-are-dropped/4074.

Currently it doesn't seem to be configurable. I'll test it later and will eventually add a note to the guide in https://github.com/pocketbase/pocketbase/discussions/537.

**Reply by cRaZy92** [reply]
Thans for reply!
Yea it doesnt really affect the connection, it works perfect, only the errors are bit annoying.
Thanks for explanation, it does make sense, if I host it myself on VPS, it should be fine? Or it should be configurable?

**Reply by ganigeorgiev** [reply]
> if I host it myself on VPS, it should be fine

Yes, if you host it on a more traditional setup then you'll have more control on the server read/write timeouts.

Fly.io unfortunately doesn't seem to have configuration for that, at least based on the comments in the above forum link.
From an older issue https://community.fly.io/t/is-it-possible-to-increase-the-timeout-to-120-sec/3007/5:

> It’s 60 seconds.
>
> Applies to both HTTP and raw TCP. They’re “idle” timeouts. **If no data is read or written within 60s, the connection is forcefully closed.** It applies for both edge connections and connections we make to your app instances.

I'll update the fly.io guide with a note about this sometime later today.

**Reply by cRaZy92** [reply]
Okay, thank you so much for this info!

**Reply by sgtpep** [reply]
@ganigeorgiev While the client re-connects, does PocketBase maintain some queue of unsent messages to deliver them after the new connect is established? Is there a chance of losing messages on reconnects?

**Reply by ganigeorgiev** [reply]
> does PocketBase maintain some queue of unsent messages to deliver them after the new connect is established?

No.

> Is there a chance to loose messages on reconnects?

Yes, if an event happens during the time the client is disconnected.

**Reply by dschissler** [reply]
@ganigeorgiev @sgtpep What about if Pocketbase could handle the [lastEventId](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent/lastEventId)?  It would need some sort of internal caching though.

**Reply by ganigeorgiev** [reply]
@dschissler It is not possible to use the lastEventId because at the moment we don't store the messages anywhere but I have it in my todo and will consider it in the future (_most likely together with allowing passing filter expr in addition to the subscription topic_).

**Reply by dschissler** [reply]
@ganigeorgiev The magic there would be if subscribing filters are the same as querying filters.  Also allowing it to somehow include the results of a query on connect based upon some variables / parameters that were sent and with lastEventId.  Of course that is a lot of work and probably with a lot of edge cases.  It seems pretty great though.

**Reply by sgtpep** [reply]
I'm relatively new to realtime stuff. One of the problems that I'm trying to solve with the current implementation of realtime in PocketBase is making sure no record updates are lost on reconnections due to timeouts and going offline.

I vaguely see how `lastEventId` could be useful for such scenarios, but until that's added to PocketBase, I found a workaround: I subscribe to `PB_CONNECT`: `pb.realtime.addEventListener('PB_CONNECT', ...)` and re-fetch the whole record(s) every time it fires to re-sync data.

Definitely, would be great to have a more straightforward way of doing that (possibly a more high-level one).

**Reply by MaCleodWalker** [reply]
@sgtpep Im trying to solve a similar issue could you provide a little more detail on what you did? Thank you.

**Comment by starredev** [comment]
Im facing exact the same issue. We are behind docker swarm and our own nginx configuration (its pretty standard tbh), seems that the connection lost after 2 minutes. 

![image](https://github.com/pocketbase/pocketbase/assets/31880091/02d8beeb-4e24-4f30-bbe0-f98aee022e53)

**Reply by ganigeorgiev** [reply]
@raizdev Please open a new discussion with more details about your use case and deployment setup.
This discussion is related to a default Fly idle timeout and your problem seems to be caused by something totally else.
For example nginx config you can refer to https://pocketbase.io/docs/going-to-production/#using-reverse-proxy (notice the `proxy_read_timeout` directive).

**Reply by starredev** [reply]
You are correct. I totally missed that part. I’m sorry it’s fixed
